src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -552,27 +552,27 @@ private void NoteFieldDefinitions()
             {
                 if (!_flags.FieldDefinitionsNoted)
                 {
-                    if (!this.IsAbstract)
-                    {
-                        var assembly = (SourceAssemblySymbol)ContainingAssembly;
+                    var assembly = (SourceAssemblySymbol)ContainingAssembly;
 
-                        Accessibility containerEffectiveAccessibility = EffectiveAccessibility();
+                    Accessibility containerEffectiveAccessibility = EffectiveAccessibility();
 
-                        foreach (var member in _lazyMembersAndInitializers.NonTypeNonIndexerMembers)
+                    foreach (var member in _lazyMembersAndInitializers.NonTypeNonIndexerMembers)
+                    {
+                        FieldSymbol field;
+                        if (!member.IsFieldOrFieldLikeEvent(out field) || field.IsConst || field.IsFixed)
                         {
-                            FieldSymbol field;
-                            if (!member.IsFieldOrFieldLikeEvent(out field) || field.IsConst || field.IsFixed)
-                            {
-                                continue;
-                            }
+                            continue;
+                        }
 
-                            Accessibility fieldDeclaredAccessibility = field.DeclaredAccessibility;
-                            if (fieldDeclaredAccessibility == Accessibility.Private)
-                            {
-                                // mark private fields as tentatively unassigned and unread unless we discover otherwise.
-                                assembly.NoteFieldDefinition(field, isInternal: false, isUnread: true);
-                            }
-                            else if (containerEffectiveAccessibility == Accessibility.Private)
+                        Accessibility fieldDeclaredAccessibility = field.DeclaredAccessibility;
+                        if (fieldDeclaredAccessibility == Accessibility.Private)
+                        {
+                            // mark private fields as tentatively unassigned and unread unless we discover otherwise.
+                            assembly.NoteFieldDefinition(field, isInternal: false, isUnread: true);
+                        }
+                        else if (!this.IsAbstract)
```

Comment:
Turned out there is a different comptability issue here as well. We don't report WRN_UnassignedInternalField (649) on unassigned internal fields, and the native csc does. I'll add a fix/test that as well.